src/nvim/getchar.c

```diff
@@ -1108,6 +1118,12 @@ static void gotchars(char_u *chars, size_t len)
       add_buff(&recordbuff, buf, 1L);
     }
   }
+
+  // remember how many chars were last recorded
+  if (Recording) {
+    last_recorded_len += num_bytes;
+  }
```

Comment:
Other than testing basic macros I can't think of any.
I had assumed there were tests for macros already in the testsuite, but I can't find them at the moment, so I'm adding some to this PR.